src/rust/engine/remote_provider/remote_provider_reapi/src/byte_store.rs

```diff
@@ -390,6 +444,139 @@ impl ByteStoreProvider for Provider {
         .map_err(|e| e.to_string())
     }
 
+    fn batch_load_supported(&self) -> bool {
+        self.batch_load_enabled
+    }
+
+    async fn load_batch(
+        &self,
+        digests: Vec<Digest>,
+        destination: &mut dyn BatchLoadDestination,
+    ) -> Result<HashMap<Digest, Result<bool, String>>, String> {
+        let mut max_batch_total_size_bytes = {
+            let capabilities = self.get_capabilities().await.map_err(|e| e.to_string())?;
+
+            capabilities
+                .cache_capabilities
+                .as_ref()
+                .map(|c| c.max_batch_total_size_bytes as usize)
+                .unwrap_or_default()
+        };
+        if max_batch_total_size_bytes == 0 || self.batch_api_size_limit < max_batch_total_size_bytes
+        {
+            max_batch_total_size_bytes = self.batch_api_size_limit;
+        }
+        if max_batch_total_size_bytes == 0 {
+            max_batch_total_size_bytes = DEFAULT_MAX_GRPC_MESSAGE_SIZE;
+        }
+        max_batch_total_size_bytes = max_batch_total_size_bytes
+            - self
+                .instance_name
+                .as_ref()
+                .cloned()
+                .unwrap_or_default()
+                .len()
+            - 10;
+
+        let mut chunks: Vec<Vec<Digest>> = Vec::new();
+        let mut current_chunk: Vec<Digest> = Vec::new();
+        let mut current_size = 0;
+
+        for digest in digests.iter() {
+            let message_size = digest.size_bytes + RPC_RESPONSE_PER_ITEM_SIZE;
+            if current_size + message_size >= max_batch_total_size_bytes {
```

Comment:
chunk size calculation now accounts for both the pants option and value from the capabilities api